mlflow/store/artifact/databricks_models_artifact_repo.py

```diff
@@ -110,12 +110,19 @@ def _get_signed_download_uri(self, path=None):
                 "API request to get presigned uri to for file under path `%s` failed with"
                 " status code %s. Response body: %s" % (path, response.status_code, response.text)
             )
-        return json_response.get("signed_uri", None)
+        return json_response.get("signed_uri", None), json_response.get("headers", None)
+
+    @staticmethod
+    def _process_raw_headers(raw_headers):
+        if raw_headers == None or len(raw_headers) == 0:
+            return {}
+        return {cur_header.get("key"): cur_header.get("value") for cur_header in raw_headers}
```

Comment:
not sure how defensive we want to be here about malformed header responses/changes to the header structure.

Reply:
`DatabricksArtifactRepo._convert_http_headers` seems more robust. Can we use that implementation? For example, `cur_header.get("key")` will return `None` if `key` isn't defined as opposed to skipping the entry.

Reply:
I can do this but I'll need to patch the universe protobuf definition of HttpHeader to match the one that the databricks artifact repo is using.  I think there is a slight difference in field naming.

Reply:
I see - no need in that case. Separate method is fine, but let's make sure it's similarly robust to issues

Reply:
I have a PR to fix the headers in universe.  I had originally made that method in DatabricksArtifactRepo static but I don't actually like how that looks and have now backed out and we just have convenience methods in both classes.  If we really want, we can set up a utils.py file and have both classes read from it, but we can do that in a follow on commit.